addons/project_account/models/project_project.py

```diff
@@ -28,6 +28,52 @@ def _get_add_purchase_items_domain(self):
             ('id', 'not in', purchase_order_line_invoice_line_ids),
         ]
 
+    def _get_revenue_items_from_purchase(self, domain, profitability_items, with_action=True):
+        account_move_lines = self.env['account.move.line'].sudo().search_fetch(
+            domain + [('analytic_distribution', 'in', self.account_id.ids)],
+            ['price_subtotal', 'parent_state', 'currency_id', 'analytic_distribution', 'move_type', 'move_id'],
+        )
+        if account_move_lines:
+            amount_invoiced = amount_to_invoice = 0.0
+            for move_line in account_move_lines:
+                revenue = 0.0
+                price_subtotal = move_line.currency_id._convert(
+                    from_amount=move_line.price_subtotal, to_currency=self.currency_id,
+                )
+                analytic_contribution = sum(
+                    percentage for ids, percentage in move_line.analytic_distribution.items()
+                    if str(self.account_id.id) in ids.split(',')
+                ) / 100.
+                expense_policy = move_line.product_id.expense_policy
+                if expense_policy == 'cost':
+                    revenue = price_subtotal * analytic_contribution
+                elif expense_policy == 'sales_price':
+                    revenue = move_line.product_id.list_price * move_line.quantity
```

Comment:
This confuses me, should it be from the reinvoiced move instead? Is this value coming out of nowhere without any journal entry linked to it?